catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/AuthenticationConfig.java

```diff
@@ -36,7 +36,8 @@ public class AuthenticationConfig extends Config {
 
   enum AuthenticationType {
     SIMPLE,
-    KERBEROS;
+    KERBEROS,
+    AWS;
```

Comment:
From my perspective, does the type seem to be AWS_AKSK, AWS_ANONYMOUS and AWS_TEMPORARY? Please correct me if I am mistaken.

Reply:
Does AWS have  anonnymous? TEMPORARY may be inaccurate.Maybe we should use token.

Reply:
@yuqi1129 

> From my perspective, does the type seem to be AWS_AKSK, AWS_ANONYMOUS and AWS_TEMPORARY? Please correct me if I am mistaken.

That's actually correct. But I'm worried that we are introducing too much values for `AuthenticationType` in this way, as we may support more storage types in the future, are you ok with this?

Reply:
@jerqi 
> Does AWS have anonnymous?

Yes, you can refer to this.
![image](https://github.com/user-attachments/assets/2c0688f7-6899-4ac2-9a8d-1752c3a369e0)


> TEMPORARY may be inaccurate.Maybe we should use token.

I use `temporary` because the name of the provider is `TemporaryAWSCredentialsProvider`, I'm also ok with the name `token`.

Reply:
> That's actually correct. But I'm worried that we are introducing too much values for `AuthenticationType` in this way, as we may support more storage types in the future, are you ok with this?

Introducing too many authentication types is indeed a problem, but `AWS` or `S3` is a service or product name NOT an authentication type, It's also not very proper to use it. @xiaozcy @jerqi Do you have any other good ideas?

Reply:
> Introducing too many authentication types is indeed a problem, but `AWS` or `S3` is a service or product name NOT an authentication type, It's also not very proper to use it. @xiaozcy @jerqi Do you have any other good ideas?

I'm also hesitant about this actually 😂 . As it's the unique authentication type of the service, there seems to be no other better way to me.

Reply:
@jerqi @jerryshao do you have any suggestions?

Reply:
To make it more clear, I think there are two ways to solve this:
- Treat these configurations in S3 as catalog properties simply. In this way, we don't have to figure out a name for the type of authentication(which also may not be appropriate). But we may have to add a property like `storage-type` to specify the actual storage system for Hadoop catalog, because we cannot do some checks and transforms for these configurations without knowing this. As more storage systems would be supported in the future, I think this might be a better way.
- Still add a new authentication type like we already did. In this case, we have to figure out a name for it. Based on the previous discussion, the candidates would be `AWS`, `AWS-AKSK`, `S3` or `S3_AKSK`. Although all of these seems not perfect, I would choose `S3_AKSK` for me, despite of the fact that we support anonymous access to a publicly accessible S3 bucket.

WDYT? @jerryshao @yuqi1129 @jerqi